tsdb/querier.go

```diff
@@ -354,8 +354,9 @@ func inversePostingsForMatcher(ctx context.Context, ix IndexReader, m *labels.Ma
 	}
 
 	res := vals[:0]
-	// If the inverse match is ="", we just want all the values.
-	if m.Type == labels.MatchEqual && m.Value == "" {
+	// If the inverse match is ="" or =~"", we just want all the values.
```

Comment:
This made me stop and think.  Is this clearer?
```suggestion
	// If the match before inversion was !="" or !~"", we just want all the values.
```

Reply:
Thanks! Addressed in latest commit.

Reply:
But it isn't, right? I'll just commit @bboreham's suggestion.